vectordraw/static/js/src/vectordraw.js

```diff
@@ -0,0 +1,607 @@
+        };
+        var default_point_style = {
+            size: 1,
+            withLabel: false,
+            color: 'pink',
+            showInfoBox: false
+        };
+        settings.points.forEach(function(point) {
+            _.defaults(point, default_point);
+            _.defaults(point.style, default_point_style);
+            point.style.name = point.name;
+            point.style.fixed = point.fixed;
+            point.style.strokeColor = point.style.color;
+            point.style.fillColor = point.style.color;
+            delete point.style.color;
+        });
+
+        return settings;
+    };
+
+    VectorDraw.prototype.render = function() {
+        // Assign the jxgboard element a random unique ID,
+        // because JXG.JSXGraph.initBoard needs it.
+        this.element.find('.jxgboard').prop('id', _.uniqueId('jxgboard'));
+        this.createBoard();
+    };
+
+    VectorDraw.prototype.createBoard = function() {
+        var id = this.element.find('.jxgboard').prop('id'),
+            self = this;
+
+        this.board = JXG.JSXGraph.initBoard(id, {
+            keepaspectratio: true,
+            boundingbox: this.settings.bounding_box,
+            axis: this.settings.axis,
+            showCopyright: false,
+            showNavigation: this.settings.show_navigation
+        });
+
+        function getImageRatio(bg, callback) {
+            $('<img/>').attr('src', bg.src).load(function(){
+                //technically it's inverse of ratio, but we need it to calculate height
+                var ratio = this.height / this.width;
+                callback(bg, ratio);
+            });
+        }
+
+        function drawBackground(bg, ratio) {
+            var height = (bg.height) ? bg.height : bg.width * ratio;
+            var coords = (bg.coords) ? bg.coords : [-bg.width/2, -height/2];
+            self.board.create('image', [bg.src, coords, [bg.width, height]], {fixed: true});
+        }
+
+        if (this.settings.background) {
+            if (this.settings.background.height) {
+                drawBackground(this.settings.background);
+            }
+            else {
+                getImageRatio(this.settings.background, drawBackground);
+            }
+        }
+
+        var noOptionSelected = true;
+
+        function renderOrEnableOption(element, idx, type, board) {
+            if (element.render) {
+                if (type === 'point') {
+                    board.renderPoint(idx);
+                } else {
+                    board.renderVector(idx);
+                }
+            } else {
+                // Enable corresponding option in menu ...
+                var option = board.getMenuOption(type, idx);
+                option.prop('disabled', false);
+                // ... and select it if no option is currently selected
+                if (noOptionSelected) {
+                    option.prop('selected', true);
+                    noOptionSelected = false;
+                }
```

Comment:
I didn't realize you would need to do this now the template code is moved to the static template.  Instead of integrating this in the loop, you could also simply enable all options using

    this.element.find('.menu option').prop('disabled', false);

and then select the first one (if necessary) with

    this.element.find('.menu option:first-child').prop('selected', true);